pkg/reconciler/cluster/cluster.go

```diff
@@ -107,7 +107,19 @@ func (c *Controller) reconcile(ctx context.Context, cluster *clusterv1alpha1.Clu
 		}.String())
 	}
 
-	if !sets.NewString(cluster.Status.SyncedResources...).Equal(groupResources) {
+	syncerInstallNeeded := false
+	if c.syncerMode == SyncerModePull {
+		upToDate, err := isSyncerInstalledAndUpToDate(ctx, client, logicalCluster, c.syncerImage)
+		if err != nil {
+			klog.Errorf("error checking if syncer needs to be installed: %v", err)
+			return err
+		}
+		if !upToDate {
```

Comment:
Is there any reason to have two variables (opposed in meaning) for one check? We could e.g.:


```go
var upToDate bool
if c.syncerMode == SyncerModePull {
    var err error 
    upToDate, err = isSyncerInstalledAndUpToDate(...
}

if !sets... || !upToDate {

}
```

Reply:
Fixed.
It looks like there is room for improvement in the install/uninstall process (for example, when going from pull mode to push mode, syncer should be uninstalled). I'll address this in another PR.

Reply:
I opened https://github.com/kcp-dev/kcp/pull/357 to improve `syncer` install/uninstall process.